godel_surface_detection/src/detection/surface_detection.cpp

```diff
@@ -433,5 +410,29 @@ namespace godel_surface_detection
 
       return name;
     }
+
+    void SurfaceDetection::filterFullCloud()
+    {
+      pcl::PointCloud<pcl::PointXYZRGB>::Ptr intermediate_cloud_ptr(new pcl::PointCloud<pcl::PointXYZRGB>);
+
+      //remove the table using the passthrough filter
+      pcl::PassThrough<pcl::PointXYZRGB> pass;
+      pass.setInputCloud(full_cloud_ptr_);
+      const std::string FILTER_DIRECTION = "z";
+      pass.setFilterFieldName (FILTER_DIRECTION);
+      //keep poin clouds with these limits
+      const double MINIMUM_DISTANCE = 0.01; // 1 cm
+      const double MAXIMUM_DISTANCE = 1.0; // 1 m
+      pass.setFilterLimits (MINIMUM_DISTANCE, MAXIMUM_DISTANCE);
+      pass.filter (*intermediate_cloud_ptr);
+
+      //downsample the full cloud using the voxelgrid filter method
+      pcl::VoxelGrid<pcl::PointXYZRGB> vox;
+      vox.setInputCloud (intermediate_cloud_ptr);
+      vox.setLeafSize (INPUT_CLOUD_VOXEL_FILTER_SIZE,
+                       INPUT_CLOUD_VOXEL_FILTER_SIZE,
+                       INPUT_CLOUD_VOXEL_FILTER_SIZE);
+      vox.filter(*process_cloud_ptr_);
+    }
```

Comment:
If this is meant to be a quick workaround for expedience then I think this should this be logged as an enhancement/issue that we can address in the future?